src/pages/Admin/Charity/Banking/RecipientDetails/useRecipientDetails.ts

```diff
@@ -0,0 +1,148 @@
+  const { id: endowment_id } = useAdminContext();
+  const [requirementsDataArray, setRequirementsDataArray] = useState<
+    RequirementsData[]
+  >([]);
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [quote, setQuote] = useState<Quote>();
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [isSubmitting, setSubmitting] = useState<boolean>(false);
+
+  const { handleError } = useErrorContext();
+
+  // const [createQuote, { isError }] = useCreateQuoteMutation();
+  // const [getAccountRequirements, { isError: isError1 }] =
+  //   useGetAccountRequirementsMutation();
+  const [getAccountRequirementsForRoute, { isError: isError1 }] =
+    useGetAccountRequirementsForRouteMutation();
+  const [postAccountRequirements, { isError: isError2 }] =
+    usePostAccountRequirementsMutation();
+  const [createRecipientAccount, { isError: isError3 }] =
+    useCreateRecipientAccountMutation();
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const sourceAmount = calculateExpectedWithdrawAmount(expectedFunds);
+        // const quote = await createQuote(targetCurrency, sourceAmount).unwrap();
+        // const newRequirements = await getAccountRequirements(quote.id).unwrap();
+        const newRequirements = await getAccountRequirementsForRoute({
+          targetCurrency,
+          sourceAmount,
+        }).unwrap();
+
+        setRequirementsDataArray(
+          newRequirements.map((item) => {
+            const data: RequirementsData = {
+              accountRequirements: item,
+              currentFormValues: getDefaultValues(item, targetCurrency),
+              refreshed: false,
+              refreshOnSubmit: item.fields.some((field) =>
+                field.group.some((group) => group.refreshRequirementsOnChange)
+              ),
+            };
+            return data;
+          })
+        );
+        setQuote(quote);
+        setLoading(false);
+      } catch (error) {
+        handleError(error, ERROR_MSG);
+        setLoading(false);
+      }
+    })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const updateDefaultValues = useCallback(
+    (formValues: FormValues) => {
+      setRequirementsDataArray((prev) => {
+        const updated = [...prev];
+        updated[selectedIndex].currentFormValues = formValues;
+        return updated;
+      });
+    },
+    [selectedIndex]
+  );
+
+  const handleSubmit = async (request: CreateRecipientRequest) => {
+    try {
+      if (!quote) {
+        throw new UnexpectedStateError("No 'quote' present.");
+      }
+
+      const requirements = requirementsDataArray.at(selectedIndex);
+      if (!requirements) {
+        throw new UnexpectedStateError("Requirements not loaded.");
+      }
+
+      setSubmitting(true);
+      // refresh requirements if necessary, see https://docs.wise.com/api-docs/api-reference/recipient#account-requirements
+      if (requirements.refreshOnSubmit && !requirements.refreshed) {
```

Comment:
@misicnenad, this would mean user would user would need to submit (`handleSubmit`) whenever to see new requirement in case one of the fields `refreshRequirementsOnChange` - what if new requirements requires refresh again ?  user would have to submit form repeatedly

I was expecting the form to refresh when I choose recipient type
<img width="316" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/741cd183-4e09-4753-8f0e-bd7e3259b8f3">

Reply:
You are right, the user *would* need to submit the form to be able to load the additional requirements (if necessary). This is because of Wise API requirements, see the below quote from [2nd paragraph in docs](https://docs.wise.com/api-docs/api-reference/recipient#account-requirements):
> Use the GET endpoint to learn what datapoints are required to send a payment to your beneficiary. As you build that form, use the POST endpoint to learn if any additional datapoints are required as a result of passing a field that has `"refreshRequirementsOnChange": true' in the GET Response. You should be posting the same recipient account payload that will be posted to `v1/accounts`.

> I was expecting the form to refresh when I choose recipient type

I would be expecting the same, but as per the explanation above, this is not possible :confused: 

To add more clarity to this logic, I added a "Check requirements" button and a short explanatory text to the form when requirement refresh is necessary. This should provide more context to our users.
I also updated the relevant logic:
- simplified check for whether refresh is needed
- separated "refresh requirements" logic from "submit form" logic

See commit https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2439/commits/52237ae2cf5cabcabfdae6fd69d715635409a230